Mastodon.js

```diff
@@ -0,0 +1,142 @@
+{
+	"translatorID": "eebd16cb-69ca-4b85-b3e2-7c603dc98562",
+	"label": "Mastodon",
+	"creator": "febrezo",
+	"target": "^https?://(mastodon\\.(social|xyz|cloud)|pawoo\\.net|mastdn.(jp|io)|switter.at)",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-07-15 15:08:26"
+}
+
+/*
+	Mastodon Translator
+	Copyright (C) 2020 Félix Brezo, felixbrezo@gmail.com
+	
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	
+	You should have received a copy of the Affero GNU General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+function detectWeb(doc, url) {
+	var urlParts = url.split('/');
+	if ( urlParts.length <= 4 ) {
+		return "webpage";
+	}	
+	return "blogPost";
+}
+
+function doWeb(doc, url) {
+	var resourceType = detectWeb(doc, url);
+	// Creating the item
+	var newItem = new Zotero.Item(resourceType);
+
+	var urlParts = url.split('/');
+	
+	var tmp_date;
+	if (resourceType == "blogPost") {
+		//Setting common data:
+		newItem.blogTitle = ZU.xpathText(doc, "//span[@class='display-name__account']/text()");
+		newItem.title = ZU.xpathText(doc, "//meta[@name='description']/@content");	
+		tmp_date = ZU.xpathText(doc, '//div[@class="detailed-status__meta"]/data/@value');
+	} else {
+		var aux = ZU.xpathText(doc, '//div[@class="public-account-bio__extra"]/text()');
+		tmp_date = aux.substring(aux.length, aux.length-8);
+		newItem.title = ZU.xpathText(doc, "//meta[@property='profile:username']/@content");		
+	}
+
+	newItem.websiteTitle = urlParts[2];
+	newItem.date = ZU.strToISO(tmp_date);
+	var tmp_author = ZU.xpathText(doc, "//strong[@class='display-name__html p-name emojify']");		
```

Comment:
This is going to be fragile this way regardless — is there more stable source for the author? — but if we're going to scrape it it'd be better to use `text()` with a CSS selector and filter for just a single class.

We apparently don't have `text()` and `attr()` documented — I thought we did — but you can see examples in other translators. Add these lines at the top of the code, below the copyright notice:

https://github.com/zotero/translators/blob/53afc969d3ea386827c627e50dfb666a41e8dfe1/ACM%20Digital%20Library.js#L37-L39

And then use it like this:

```javascript
 let reviewedTitle = text(doc, '.article-reviewed-item-title'); 
```

Reply:
Yes it is, the `<h1>` tag can be used. I have updated it.

```
<h1>
Félix Brezo
<small>
@febrezo@mastodon.social

</small>
</h1>
```

I noticed that the error in the first name was caused because it was not properly grabbed.